app/models/katello/concerns/redhat_extensions.rb

```diff
@@ -72,8 +72,8 @@ def distribution_repositories(host)
 
         if content_view && lifecycle_environment
           Katello::Repository.in_environment(lifecycle_environment).in_content_views([content_view]).
-              where(:distribution_version => host.os.release,
-                    :distribution_arch => host.architecture.name,
+              where(:distribution_version => host.try(:os).try(:release),
```

Comment:
In case `host.os == nil`, this would render to `where distribution_version = null` isn't it?
In this case the list will always remain empty AFAIU...